src/extension/vectortiles/src/main/java/org/geotools/renderer/lite/VectorMapRenderUtils.java

```diff
@@ -443,4 +445,33 @@ private static boolean isFeatureTypeStyleActive(FeatureType ftype, FeatureTypeSt
                                 || FeatureTypes.isDecendedFrom(
                                         ftype, null, fts.getFeatureTypeName())));
     }
+
+    /**
+     * This function executes transformations on the vector features e.g. point stacker.
+     *
+     * @param layer the layer to process
+     * @param paintArea the area that is painted on
+     * @param mapScale the current map scale
+     * @param schema the schema of the features
+     * @param features the feature collection typically points
+     * @return the transformed feature collection
+     * @throws IOException
+     */
+    public static FeatureCollection<?, ?> transformIfNecessary(
+            Layer layer,
+            Rectangle paintArea,
+            double mapScale,
+            FeatureType schema,
+            FeatureCollection<?, ?> features)
+            throws IOException {
+
+        List<LiteFeatureTypeStyle> featureTypeStyles =
+                VectorMapRenderUtils.getFeatureStyles(layer, paintArea, mapScale, schema);
+        if (!featureTypeStyles.isEmpty() && featureTypeStyles.get(0).transformation != null) {
```

Comment:
A layer can have multiple FTS attached and one can be transformed while the others are not, or have different transformations per FTS. 
If I understand how the module works (mind, not the original creator nor its maintainer) for this case to be supported the layer should be split in multiple ones, I'm guessing by checking the FTSs and grouping them by uniform transform.

Reply:
@aaime I agree that it might be possible that there are multiple transformation. Nevertheless the patch already enhances the behavior of geoserver a  lot. Currently the transformations are not applied at all.

Reply:
It's not up to me to decide if that's ok or not, that's the responsibility of the module maintainer. If it's ok to be included as is, I'd add a small note in the documentation that the RT must be in the first FTS and others might be ignored (but again, not my call).

Reply:
@aaime  thanks a lot for reviewing. Let's see what the maintainer says.

Reply:
This should apply transforms to each FeatureTypeStyle, not just the first.